esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2298,6 +2264,66 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
     return true;
 }
 
+void CFileSprayEx::checkDZScopeAccessAndSetSpraySourceDFUFileSpec(IEspContext &context, const char *srcPlane, const char *srcHost,
+    const char *srcFile, MemoryBuffer &srcXML, IDFUfileSpec *srcDFUfileSpec)
+{
+    if(srcXML.length() == 0)
+    {
+        RemoteMultiFilename rmfn;
+        SocketEndpoint ep(srcHost);
+        if (ep.isNull())
+            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Cannot resolve source network IP from %s.", srcHost);
+
+        rmfn.setEp(ep);
+        StringBuffer fnamebuf(srcFile);
+        fnamebuf.trim();
+        rmfn.append(fnamebuf.str());    // handles comma separated files
+        srcDFUfileSpec->setMultiFilename(rmfn);
+
+        ForEachItemIn(i, rmfn)
+        {
+            StringBuffer path;
+            rmfn.item(i).getLocalPath(path);
+
+            //A spray source could be: '/path/*'.
+            StringBuffer dirPath, dirTail;
+            splitFilename(path, &dirPath, &dirPath, &dirTail, &dirTail);
+
+            const char* s = streq(dirTail.str(), "*") ? dirPath : path;
+            SecAccessFlags permission = getDropZoneScopePermissions(context, srcPlane, s, srcHost);
```

Comment:
incidental issue (not for this PR) : I think there is code in getDropZoneScopePermissions(IEspContext&, which is adding a separator prefix to the relative path, that it shouldn't - possibly as a workaround for a bug since fixed in setPlaneExternal which may have relied on one. Please check/fix, but not in this JIRA/PR.

Reply:
>I read the code inside the getDropZoneScopePermissions(). The code does not add a prefix to a relative path

This code:
```
        const char* p = dropZonePath + strlen(prefix);
        if (!*p || !isPathSepChar(p[0]))
            addPathSepChar(s);
        s.append(p);
        dropZonePath = s.str();
```

the addPathSepChar(s) is ensuring the relative path is prefixed with '/' if the dropZonePath is same as prefix or if the point of similarity is not a path separator. But the relative path passed to queryDistributedFileDirectory().getDropZoneScopePermissions should not be prefixed with a '/'.

NB: there's a separate bug here as well, e.g. if the named dropzone prefix is /data/mydropzone and getDropZoneScopePermissions is passed a dropZonePath of /data/mydropzone-notreally/dir1/file1, then hasPrefix will return true, and it will cause this function to construct a relative path like "/-notreally/dir1/file1".
Fix this in a separate JIRA/PR though.

Reply:
>But, the scope name sent from dfuserver to the getPermissionsLDAP() is: 'file::ip::absolute_path::filename'.

Yes, that is why HPCC-29246 was opened, so it is still needed.
However, as noted in that JIRA's description, we need to ensure backward compatibility with existing LDAP servers.
We should check plane::<name>::dir1::dir2::file1, but we will need to have some code in place to fallback to check the file:: form too if that fails.

Reply:
> queryDistributedFileDirectory().getDropZoneScopePermissions should not be prefixed with a '/'.
I understand what you mean now. Then, I should make sure that the path does not have a leading '/'.

Reply:
yes, it should really, but please don't change in this PR.
Keep changes to a minimum, small review, then tackle other issues discovered separately and individually.

Reply:
https://track.hpccsystems.com/browse/HPCC-29316